src/models/m_wikiconcept.erl

```diff
@@ -290,6 +299,92 @@ filter_lang(#trans{ tr = Tr }, Context) ->
     #trans{ tr = Tr2 }.
 
 
+%% @doc Handle the merge of two keywords. Merge all connected concepts.
+-spec rsc_merge(WinnerId, LoserId, Context) -> ok when
+    WinnerId :: m_rsc:resource_id(),
+    LoserId :: m_rsc:resource_id(),
+    Context :: z:context().
+rsc_merge(WinnerId, LoserId, Context) ->
+    case z_db:q("
+        update wikiconcept
+        set keyword_id = $1
+        where keyword_id = $2",
+        [ WinnerId, LoserId ],
+        Context)
+    of
+        0 ->
+            ok;
+        Count when is_integer(Count) ->
+            publish_update(WinnerId, Context)
+    end.
+
+%% @doc Find the keyword best matching to the given concept.
+%% First tries to find a keyword in the ascendants of a concept,
+%% then tries to find it in the descendents.
+-spec find_keyword(ConceptID, Context) -> {ok, KeywordRscId} | {error, Reason} when
+    ConceptID :: binary() | string(),
+    Context :: z:context(),
+    KeywordRscId :: m_rsc:resource_id(),
+    Reason :: term().
+find_keyword(ConceptID, Context) ->
+    case get_concept(ConceptID, Context) of
+        {ok, #{
+            <<"keyword_id">> := RscId
+        }} when is_integer(RscId) ->
+            {ok, RscId};
+        {ok, #{
+            <<"wikidata_id">> := WikidataID,
+            <<"wikidata_ancestor_ids">> := Ancestors
+        }} ->
+            case find_nearest_ancestor(Ancestors, Context) of
+                {ok, RscId} ->
+                    {ok, RscId};
+                {error, enoent} ->
+                    find_nearest_descendent(WikidataID, Context)
+            end;
+        {error, _} = Error ->
+            Error
+    end.
+
+find_nearest_ancestor(undefined, _Context) ->
+    {error, enoent};
+find_nearest_ancestor([], _Context) ->
+    {error, enoent};
+find_nearest_ancestor(Ancestors, Context) ->
+    case z_db:q1("
+        select keyword_id
+        from wikiconcept
+        where keyword_id is not null
+          and wikidata_id in (select(unnest($1::character varying[])))
+        order by level desc
+        limit 1",
+        [ Ancestors ],
+        Context)
+    of
+        undefined ->
+            {error, enoent};
+        RscId ->
+            {ok, RscId}
+    end.
+
+find_nearest_descendent(WikidataID, Context) ->
+    case z_db:q1("
+        select keyword_id
+        from wikiconcept
+        where keyword_id is not null
+          and wikidata_ancestor_ids && $1
+        order by level asc
+        limit 1",
+        [ [ WikidataID ] ],
```

Comment:
Similarly to above, I think that this is easier to read as:
```suggestion
        where keyword_id is not null
          and $1 = ANY(wikidata_ancestor_ids)
        order by level asc
        limit 1",
        [ WikidataID ],
```

Reply:
Oh, indeed, after [a quick online search](https://stackoverflow.com/questions/4058731/can-postgresql-index-array-columns/29245753#29245753) it seems that `ANY` indeed doesn't use the GIN index.

Never mind then, this is supposed to be considerably more performant, so I agree that it's best left as-is.